plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -330,9 +330,16 @@ private static String quotedName(String name)
         return '"' + name.replace("\"", "\"\"") + '"';
     }
 
-    public static boolean canEnforceColumnConstraintInAllSpecs(TypeOperators typeOperators, Table table, IcebergColumnHandle columnHandle, Domain domain)
+    public static boolean canEnforceColumnConstraintInAllSpecs(
+            TypeOperators typeOperators,
+            Table table,
+            IcebergColumnHandle columnHandle,
+            Domain domain,
+            Set<Integer> partitionSpecIds)
     {
         return table.specs().values().stream()
+                // If the table is empty partitionSpecIds may be empty
+                .filter(partitionSpec -> partitionSpecIds.isEmpty() || partitionSpecIds.contains(partitionSpec.specId()))
```

Comment:
if the table is empty, we can enforce all the predicates, so we don't need any special treatment for that case

the special treatment for empty partitionSpecIds is worrying (it calls for the param to be Optional or something), but since it's seems not be needed, let's remove it